custom_components/localtuya/pytuya/__init__.py

```diff
@@ -223,48 +260,22 @@ def detect_available_dps(self):
         # in the ranges [1-25] and [100-110] need to split the bruteforcing in
         # different steps due to request payload limitation (max. length = 255)
         detected_dps = {}
-
-        # dps 1 must always be sent, otherwise it might fail in case no dps is found
-        # in the requested range
-        self.dps_to_request = {"1": None}
-        self.add_dps_to_request(range(2, 11))
-        try:
-            data = self.status()
-        except Exception as e:
-            print("Failed to get status: [{}]".format(e))
-            raise
-        detected_dps.update(data["dps"])
-
-        if self.dev_type == "type_0a":
-            return detected_dps
-
-        self.dps_to_request = {"1": None}
-        self.add_dps_to_request(range(11, 21))
-        try:
-            data = self.status()
-        except Exception as e:
-            print("Failed to get status: [{}]".format(e))
-            raise
-        detected_dps.update(data["dps"])
-
-        self.dps_to_request = {"1": None}
-        self.add_dps_to_request(range(21, 31))
-        try:
-            data = self.status()
-        except Exception as e:
-            print("Failed to get status: [{}]".format(e))
-            raise
-        detected_dps.update(data["dps"])
-
-        self.dps_to_request = {"1": None}
-        self.add_dps_to_request(range(100, 111))
-        try:
-            data = self.status()
-        except Exception as e:
-            print("Failed to get status: [{}]".format(e))
-            raise
-        detected_dps.update(data["dps"])
-        #        print("DATA IS [{}] detected_dps [{}]".format(data,detected_dps))
+        ranges = [(2, 11), (11, 21), (21, 31), (100, 111)]
+
+        for dps_range in ranges:
+            # dps 1 must always be sent, otherwise it might fail in case no dps is found
+            # in the requested range
+            self.dps_to_request = {"1": None}
+            self.add_dps_to_request(range(*dps_range))
+            try:
+                data = self.status()
+            except Exception as e:
+                _LOGGER.warning("Failed to get status: [{}]", e)
```

Comment:
All the warning and error messages for Exceptions trigger the following errors:
Traceback (most recent call last):
  File "/usr/lib/python3.7/logging/__init__.py", line 1034, in emit
    msg = self.format(record)
  File "/usr/lib/python3.7/logging/__init__.py", line 880, in format
    return fmt.format(record)
  File "/usr/lib/python3.7/logging/__init__.py", line 619, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.7/logging/__init__.py", line 380, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "./tuyadebug/test.py", line 42, in <module>
    localtuya.devicePrint(DEVID, DEVIP, DEVKEY, DEVVERS)
  File "/root/tuyadebug/localtuya/__init__.py", line 162, in devicePrint
    err,data,dev_type = deviceInfo(deviceid, ip, key, vers)
  File "/root/tuyadebug/localtuya/__init__.py", line 108, in deviceInfo
    data = d.detect_available_dps()
  File "/root/tuyadebug/localtuya/pytuya/__init__.py", line 274, in detect_available_dps
    _LOGGER.error("Failed to get status: [{}]", e)
Message: 'Failed to get status: [{}]'
Arguments: (TypeError('Object of type bytes is not JSON serializable'),)

Reply:
Bad formatting of course, will fix.